### PR TITLE
refactor(admin): move user management into the settings sub-sidebar

### PR DESCRIPTION
**Pull request by muzzamil-rx**
### Overview

The standalone **Users icon** (id `"manage"`) has been removed from the main icon rail.
Users, User Fields, Roles, and any plugin collections previously under the **"users"** section are now grouped under a new **User Management** section at the top of the **Settings sub-sidebar**.

Navigation updates:

* Visiting `/admin/users` or `/admin/security/roles` now highlights the **Settings icon**
* Automatically opens the **Settings sub-sidebar**
* These routes are now treated as part of Settings across navigation and UI

Breadcrumbs:

* Users and Roles pages now appear under a **Settings parent breadcrumb**

Permission handling:

* Roles with only `read-users` or `read-roles` permissions:

  * Still see the **Settings icon**
  * Land directly on `/admin/users` or Roles
  * No longer blocked by `manage-settings` guard on the General page

---

### Summary

Moves **User Management (Users, User Fields, Roles)** into the Settings section, consolidating all admin configuration under one place.

* Removes redundant **"manage"** main-menu category
* Integrates Users/Roles into Settings navigation, active state, and breadcrumbs
* No changes to routes, permissions, or backend behavior

👉 This is a **pure admin UI/navigation update**

---

### Type of Change

* [x] Bug fix (non-breaking change that fixes an issue)
* [x] New feature (non-breaking change that adds functionality)
* [ ] Breaking change
* [ ] Documentation update
* [ ] Refactor / chore

---

### Related Issues

None

---

### Changeset

* Added changeset: `.changeset/admin-users-section-into-settings.md`
* Applied **patch version bump** (alpha lockstep across packages)

---

### Test Plan

* `pnpm lint` → ✅ clean (eslint --max-warnings 0)
* `pnpm check-types` → ✅ clean (tsc --noEmit)
* `pnpm build` → ✅ success (17/17 turbo tasks passed)

Manual testing:

* Verified navigation behavior
* Verified icon highlighting
* Verified permissions edge cases

Unit tests:

* `resolve-item-href` (8 tests)
* `sidebar-landing` (12 tests)
* `webhook-nav` (3 tests)

 Total: **23 tests — all passing**

---

### Checklist

* [x] Followed CONTRIBUTING guidelines
* [x] Used Conventional Commits
* [x] Targeted `main` branch
* [x] Updated relevant documentation

---

### Screenshots / Recordings

None yet

---

### Notes for Reviewers

* `pages/registry.ts` remains unchanged (no section/category logic exists there)
* Navigation grouping is handled via updated **nav constants**

Edge case handled:

* Users with limited permissions (users-only / roles-only):

  * Still see Settings icon (`hasSettingsSection`)
  * Redirect correctly to `/admin/users` instead of blocked General page

Legacy cleanup:

* Updated unused `SIDEBAR_NAVIGATION / SidebarNavigation`
* No functional impact (not mounted anywhere)

## Summary by CodeRabbit

* **Navigation**
  * Moved **Users**, **Roles**, and **Custom Fields** under **Settings**.
  * Removed the standalone **Users** sidebar section.
  * Updated sidebar highlighting for user-management pages and plugin entries.
  * **Settings** now opens the most relevant accessible user-management page.

* **Breadcrumbs**
  * Added **Dashboard** and **Settings** breadcrumbs to user and role management pages.
  * **Settings** breadcrumb destinations now adapt based on permissions.

### CODE REVIEW

File: packages/admin/src/components/features/dashboard/SidebarNavigation.tsx [hunk 1]
```diff
@@ -275,10 +275,11 @@ export function SidebarNavigation({
...
-  // Group items by category (using filtered items)
+  // Group items by category (using filtered items). Users/Roles/Custom Fields
+  // are now filed under "settings" (User Management moved into Settings), so
+  // there is no separate "users" group anymore.
   const mainNavItems = filteredItems.filter(item => item.category === "main");
   const mediaItems = filteredItems.filter(item => item.category === "media");
-  const userItems = filteredItems.filter(item => item.category === "users");
   const settingsItems = filteredItems.filter(
     item => item.category === "settings"
   );
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/components/features/dashboard/SidebarNavigation.tsx:285]: Thanks for flagging. `SidebarNavigation` is legacy/unmounted — there are zero render sites for it or its only consumer `MobileSidebarDrawer` in the live app (the active sidebar is `DualSidebar`). So the missing User Management render path has no runtime effect. The live Settings sub-sidebar (`SubSidebarContent.tsx`) does render Users/User Fields/Roles under a User Management group. No code change needed here; leaving the dead file as-is to avoid churn in unmounted code.

File: packages/admin/src/components/features/role-management/breadcrumbs/index.tsx [hunk 2]
```diff
@@ -26,8 +26,11 @@ const PAGE_LABELS = {
 export function RoleBreadcrumbs({ currentPage }: RoleBreadcrumbsProps) {
   const currentLabel = PAGE_LABELS[currentPage];
 
+  // Roles live under the Settings section (User Management) now, so the trail
+  // nests under a "Settings" parent crumb before the "Roles" item.
   const items: BreadcrumbItem[] = [
     { label: "Dashboard", href: ROUTES.DASHBOARD, isDashboard: true },
+    { label: "Settings", href: ROUTES.SETTINGS },
   ];
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/components/features/role-management/breadcrumbs/index.tsx:41]: Good catch — fixed in 881bd6e6. The Settings crumb in `UserBreadcrumbs` and `RoleBreadcrumbs` now resolves to a reachable Settings subpage via a small inline check (`manage-settings ? ROUTES.SETTINGS : ROUTES.USERS` / `ROUTES.SECURITY_ROLES`) so a users-only or roles-only role isn't pointed at the guarded General page. Kept it inline per the repo's minimal-change preference rather than introducing a shared resolver.

File: packages/admin/src/constants/navigation.ts [hunk 3]
```diff
@@ -116,26 +116,26 @@ export const SIDEBAR_NAVIGATION: SidebarNavigation = [
...
-    category: "users",
+    category: "settings",
     requiredPermission: "read-roles",
   },
   {
     title: "Custom Fields",
     href: ROUTES.USERS_FIELDS,
     icon: SlidersHorizontal,
-    category: "users",
+    category: "settings",
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/constants/navigation.ts:138]: Same as the sibling comment — `SidebarNavigation` (the only consumer of `showSettingsSection`) is unmounted legacy code, so this has no runtime effect. The live nav (`DualSidebar`/`SubSidebarContent`) already extends its Settings visibility to `canViewUsers`/`canViewRoles` and renders the User Management group. No change.

File: packages/admin/src/components/features/user-management/breadcrumbs/index.tsx [hunk 4]
```diff
@@ -35,8 +35,11 @@ const PAGE_LABELS = {
 export function UserBreadcrumbs({ currentPage }: UserBreadcrumbsProps) {
   const currentLabel = PAGE_LABELS[currentPage];
 
+  // User Management lives under the Settings section now, so the trail nests
+  // under a "Settings" parent crumb before the "Users" item.
   const items: BreadcrumbItem[] = [
     { label: "Dashboard", href: ROUTES.DASHBOARD, isDashboard: true },
+    { label: "Settings", href: ROUTES.SETTINGS },
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/components/features/user-management/breadcrumbs/index.tsx]: Valid — fixed in 881bd6e6. The User Management breadcrumb's Settings crumb now points at `ROUTES.USERS` for users-only roles (`manage-settings ? ROUTES.SETTINGS : ROUTES.USERS`) instead of the guarded `/admin/settings`, so it's no longer a dead parent link.

File: packages/admin/src/components/layout/sidebar/DualSidebar.tsx [hunk 5]
```diff
@@ -85,7 +85,6 @@ export function DualSidebar({ isMobile }: DualSidebarProps = {}) {
       singles: "singles",
       media: "media",
       plugins: "plugins",
-      manage: "users",
       settings: "settings",
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/components/layout/sidebar/DualSidebar.tsx:88]: Good catch — fixed in 881bd6e6. In `DualSidebar`'s standalone-plugin anchor resolution, a legacy `after: "users"` is now normalized to `"settings"` (where User Management moved), so those plugin icons anchor next to Settings instead of falling through to the end of the rail.

File: .changeset/admin-users-section-into-settings.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+"@nextlyhq/storage-uploadthing": patch
+"@nextlyhq/storage-vercel-blob": patch
+"@nextlyhq/plugin-form-builder": patch
+"@nextlyhq/plugin-page-builder": patch
+"@nextlyhq/plugin-sdk": patch
+"@nextlyhq/eslint-config": patch
+"@nextlyhq/prettier-config": patch
+"@nextlyhq/telemetry": patch
+"@nextlyhq/tsconfig": patch
+---
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply .changeset/admin-users-section-into-settings.md:23]: Not an issue here: the repo uses a Changesets `fixed` group (see `.changeset/config.json`) that lists `@nextlyhq/blocks-engine`, `@nextlyhq/plugin-seo`, and all other lockstep packages. At version time the `fixed` group is bumped together regardless of which packages an individual changeset names, so nothing gets left behind. This matches every existing changeset in the repo (e.g. `admin-publish-button-gating.md`, `admin-version-diff-view.md`), none of which enumerate those packages either.

**Comment by muzzamil-rx** [comment]
Addressed the review feedback in `881bd6e6`.

**Fixed (2 real issues):**
- **Breadcrumb dead parent link** (CodeRabbit + Codex): the Settings crumb on Users/Roles pages pointed at `/admin/settings`, which is `manage-settings`-guarded — a users-only or roles-only role would get bounced. Now resolves to a reachable subpage inline (`manage-settings ? ROUTES.SETTINGS : ROUTES.USERS`/`ROUTES.SECURITY_ROLES`). Files: `user-management/breadcrumbs`, `role-management/breadcrumbs`.
- **`after: "users"` standalone-plugin anchor orphaned** (Codex): legacy plugins anchoring after the removed Users icon now anchor next to Settings. File: `DualSidebar.tsx`.

**No change needed (4 comments):**
- CodeRabbit's two `SidebarNavigation` concerns target **unmounted legacy code** (zero render sites) — no runtime effect; the live `DualSidebar`/`SubSidebarContent` already renders User Management under Settings and extends Settings visibility to `canViewUsers`/`canViewRoles`.
- Codex's changeset concern is covered by the repo's Changesets `fixed` group, which bumps all lockstep packages together at version time (matches every existing changeset).

`pnpm check-types`, `pnpm lint`, and the sidebar unit tests all pass.

File: packages/admin/src/pages/dashboard/roles.tsx [hunk 7]
```diff
@@ -43,6 +43,8 @@ const RolesPage: React.FC = () => {
           <Breadcrumbs
             items={[
               { label: "Dashboard", href: ROUTES.DASHBOARD, isDashboard: true },
+              // Roles live under Settings (User Management) now.
+              { label: "Settings", href: ROUTES.SETTINGS },
```
[the remark this answers is not in the document]
Comment (muzzamil-rx) [reply packages/admin/src/pages/dashboard/roles.tsx]: Good catch — fixed in 048788a2. The role list page builds its breadcrumbs inline (it doesn't use `RoleBreadcrumbs`), so it still pointed at `ROUTES.SETTINGS`. It now uses the same permission-aware target inline (`manage-settings ? ROUTES.SETTINGS : ROUTES.SECURITY_ROLES`), so a `read-roles`-only user lands on Roles instead of the guarded General page. All three breadcrumbs (UserBreadcrumbs, RoleBreadcrumbs, and this inline one) are now consistent.

**Review by aqib-rx — APPROVED** [review]
